ui/src/services/user.js

```diff
@@ -27,10 +27,34 @@ class UserService {
       .patch(`/users/${username}`, updates)
       .then((response) => response.data);
   }
-
-  deleteUser(username) {
+/*
+ // Existing user deletion: soft delete
+  softDeleteUser(username) {
     return api.delete(`/users/${username}`).then((response) => response.data);
   }
+
+  // New: Hard delete a user
+  hardDeleteUser(username) {
+    return api.delete(`/users/${username}/delete`).then((response) => response.data);
+  }
+    */
+
+  // Unified deleteUser function
+  deleteUser(username, isHardDelete = false) {
+    const queryParam = isHardDelete ? "?hard_delete=true" : ""; 
```

Comment:
The query parameters can probably be provided as an object argument to the `api.delete` method, and shouldn't need to be set as a string like `?hard_delete=true`.